src/llm/llm.service.ts

```diff
@@ -26,28 +26,48 @@ import { ConversationState } from '../database/models/conversation-state.model';
 import { InjectModel } from '@nestjs/sequelize';
 import { TRIAL_MAX_MESSAGE_PER_CONVERSATION_COUNT } from '../lib/utils/slack-constants';
 import { Md } from 'slack-block-builder';
-import { encryptForLogs } from '../lib/utils/encryption';
+import { encrypt } from '../lib/utils/encryption';
 import { formatToOpenAITool } from '@langchain/openai';
 import { isEqual } from 'lodash';
 import { SOFT_RETENTION_DAYS } from '../lib/constants';
+import { getSlackMessageUrl } from '@quix/lib/utils/slack';
+import { SlackWorkspace } from '@quix/database/models';
 import slackify = require('slackify-markdown');
 
 @Injectable()
 export class LlmService {
   private readonly logger = new Logger(LlmService.name);
   constructor(
+    @InjectModel(SlackWorkspace)
+    private readonly slackWorkspaceModel: typeof SlackWorkspace,
     private readonly llmProvider: LlmProviderService,
     private readonly tool: ToolService,
     @InjectModel(ConversationState)
     private readonly conversationStateModel: typeof ConversationState
   ) {}
 
-  private enhanceMessagesWithToolContext(
+  private async enhanceMessagesWithToolContext(
     previousMessages: LLMContext[],
-    lastToolCalls: ConversationState['last_tool_calls']
-  ): LLMContext[] {
+    lastToolCalls: ConversationState['last_tool_calls'],
+    channelId: string,
+    threadTs?: string,
+    domain?: string
+  ): Promise<LLMContext[]> {
     const enhancedPreviousMessages = [...previousMessages];
 
+    if (threadTs) {
+      const slackUrl = getSlackMessageUrl({
+        slackDomain: domain || '',
+        channelId,
+        messageExternalId: threadTs
+      });
```

Comment:
We can add the link even when threadTs is absent, the thread might get new messages in the future. Instead of this check add a check on domain being null instead
```suggestion
    if (domain) {
      const slackUrl = getSlackMessageUrl({
        slackDomain: domain,
        channelId,
        messageExternalId: threadTs
      });
```